src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2389,25 +2395,42 @@ private static void AddNestedTypesToDictionary(Dictionary<string, ImmutableArray
             }
         }
 
-        private class MembersAndInitializersBuilder
+        private sealed class MembersAndInitializersBuilder
         {
             public ArrayBuilder<Symbol> NonTypeNonIndexerMembers { get; set; } = ArrayBuilder<Symbol>.GetInstance();
             public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> StaticInitializers = ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>>.GetInstance();
             public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> InstanceInitializers = ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>>.GetInstance();
             public readonly ArrayBuilder<SyntaxReference> IndexerDeclarations = ArrayBuilder<SyntaxReference>.GetInstance();
             public RecordDeclarationSyntax? RecordDeclarationWithParameters;
             public ArrayBuilder<FieldOrPropertyInitializer.Builder>? InstanceInitializersForRecordDeclarationWithParameters;
+            public bool? IsNullableEnabledForInstanceConstructorsAndFields;
+            public bool? IsNullableEnabledForStaticConstructorsAndFields;
 
-
-            public MembersAndInitializers ToReadOnlyAndFree()
+            public MembersAndInitializers ToReadOnlyAndFree(bool isNullableEnabledForProject)
             {
                 return new MembersAndInitializers(
                     NonTypeNonIndexerMembers.ToImmutableAndFree(),
                     ToReadonlyAndFree(StaticInitializers, out int staticInitializersSyntaxLength),
                     ToReadonlyAndFree(InstanceInitializers, out int instanceInitializersSyntaxLength),
                     IndexerDeclarations.ToImmutableAndFree(),
                     staticInitializersSyntaxLength,
-                    instanceInitializersSyntaxLength);
+                    instanceInitializersSyntaxLength,
+                    isNullableEnabledForInstanceConstructorsAndFields: IsNullableEnabledForInstanceConstructorsAndFields ?? isNullableEnabledForProject,
```

Comment:
>isNullableEnabledForProject [](start = 124, length = 27)

Could you please elaborate why falling back to project setting is appropriate?

Reply:
This is simply to avoid requiring the consumer of the properties to check the project context.

---
In reply to: [546782898](https://github.com/dotnet/roslyn/pull/49953#discussion_r546782898) [](ancestors = 546782898)

Reply:
> If there are no syntax spans for constructors or fields, or if the syntax spans include #nullable restore but not #nullable enable, then the project context is appropriate.

Haven't we already accounted for the project setting when we were figuring out value for each specific span? And if there are no constructors and fields declared, what is there to analyze?

---
In reply to: [547800998](https://github.com/dotnet/roslyn/pull/49953#discussion_r547800998) [](ancestors = 547800998,546782898)